internal/healthchecks/linux.go

```diff
@@ -20,9 +20,23 @@ package healthchecks
 import (
 	"errors"
 	"net"
+	"os/exec"
 	"syscall"
 )
 
+func isSubagentActive(subagent string) (bool, error) {
+	_, err := exec.Command("systemctl", "is-active", "--quiet", subagent).Output()
+	if err != nil {
+		// if the service doesn't exist return false with no error
```

Comment:
What happens in the case the `service exists`, but it is not running.

Reply:
The command exits with status code 3 so the code will run as if the service doesn't exists.